src/client_side_request.cc

```diff
@@ -484,10 +487,13 @@ ClientRequestContext::hostHeaderVerify()
     char *hostB = xstrdup(host);
     host = hostB;
     if (host[0] == '[') {
-        // IPv6 literal.
-        portStr = strchr(hostB, ']');
-        if (portStr && *(++portStr) != ':') {
-            portStr = nullptr;
+        // IPv6 literal: de-bracket and detect optional port
+        auto rb = strchr(hostB, ']');
+        if (rb) {
+            if (rb[1] == ':')
+                portStr = rb + 1;   // points to ':'
+            *rb = '\0';             // drop closing bracket
+            host = hostB + 1;       // drop opening bracket
         }
```

Comment:
> PR description: Previously, hostHeaderVerify() would incorrectly debracket addresses such as `[0:1]:6543`.

Just to clarify, AFAICT, official code does not debracket addresses at all. Official code finds the port number after the brackets, and splits input at `:`, but the `host` variable keeps its original brackets. This PR removes brackets from `host`. Is my summary accurate?